### PR TITLE
add API to force a checkpoint address

### PR DESCRIPTION
**Pull request by no2chem**
This PR adds an API to force the checkpoint writer to take the snapshot
at a specified address by calling setSnapshotAddress() by default,
the latest token is obtained from the sequencer.

### CODE REVIEW

File: runtime/src/main/java/org/corfudb/runtime/CheckpointWriter.java [hunk 1]
```diff
@@ -239,10 +243,15 @@ public long finishCheckpoint() {
         return endAddress;
     }
 
-    public static long startGlobalSnapshotTxn(CorfuRuntime rt) {
-        TokenResponse tokenResponse =
-                rt.getSequencerView().nextToken(Collections.EMPTY_SET, 0);
-        long globalTail = tokenResponse.getToken().getTokenValue();
+    public static long startGlobalSnapshotTxn(CorfuRuntime rt, Long forceAddress) {
```
Comment (Maithem) [inline runtime/src/main/java/org/corfudb/runtime/CheckpointWriter.java]: If this method depends on a member field, do we still want this method to be static?

File: runtime/src/main/java/org/corfudb/runtime/MultiCheckpointWriter.java [hunk 2]
```diff
@@ -74,6 +77,7 @@ public long appendCheckpoints(CorfuRuntime rt, String author,
                                 .getSerializer();
                 cpw.setSerializer(serializer);
                 cpw.setPostAppendFunc(postAppendFunc);
+                cpw.setSnapshotAddress(snapshotAddress);
```
Comment (Maithem) [inline runtime/src/main/java/org/corfudb/runtime/MultiCheckpointWriter.java]: what happens if the snapshotaddress is -1 ?

**Comment by slfritchie** [comment]
To be of value, the snapshot's address needs to find its way into the START_LOG_ADDRESS metadata.  Commit cd76ad7 is supposed to do that, please double-check that I'm fetching the address correctly.

**Comment by corfudb-performance** [comment]
Results automatically generated by _CorfuDB Benchmark Framework_ to assess the performance of this pull request for commit **cd76ad7**.

*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 10 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 10 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 10 threads, Disk mode
 
 An interactive dashboard with Pull Request Performance Metrics for ALL cluster types and numbers of threads in run, is available at:
 [Pull Request #673 Graphs ](https://snapshot.raintank.io/dashboard/snapshot/1STj66olLcuviuQSpvwfhFoyluF2KaYV)

**Pull request closed without merging**